src/components/journeys/JourneyInstancesDataTable/getStaticColumns.tsx

```diff
@@ -75,12 +72,14 @@ export const getStaticColumns = (
   ];
 
   // Add header names
-  return staticColumns
-    .filter((staticColumn) =>
-      Object.keys(columnNames).includes(staticColumn.field)
-    )
-    .map((col) => ({
-      headerName: columnNames[col.field],
-      ...col,
-    }));
+  return staticColumns.map((col) => ({
+    renderHeader: () => (
+      <div className="MuiDataGrid-columnHeaderTitle">
+        <FormattedMessage
+          id={`pages.organizeJourneyInstances.columns.${col.field}`}
+        />
+      </div>
+    ),
+    ...col,
+  }));
```

Comment:
I think this is still unnecessary. Why have a special loop for adding the names, instead of just adding the names as part of the definition further up in the file?

When it's separated like this, it means I have to look in two places if I'm wondering why a column looks the way it does. It also means tying the message ID in the frontend YAML files to the field name from the backend API, which I think is an unnecessary dependency (and has caused the linter to complain in that YAML files, prompting you to disable that rule altogether in that file).

I don't think it's never ok to do something like using API fields as keys for messages, but it comes at a cost (linting, dependencies, etc) and I don't see the benefit in this case.

I think this would be much better solved the way it was in the `ViewsListTable` component, here: https://github.com/zetkin/app.zetkin.org/blob/670bf0960c1a1addf1ad9606ad0d7e012771e411/src/components/views/ViewsListTable.tsx#L63-L70


Is there any reason the same approach should not be used here?

Reply:
For some reason you're looking at an outdated version of the PR. This is gone now